drivers/dma/dma_stm32_v3.c

```diff
@@ -270,11 +269,69 @@ static int dma_stm32_validate_transfer_sizes(struct dma_config *config)
 	return 0;
 }
 
+static void dma_stm32_configure_linked_list(uint32_t id, struct dma_config *config,
+					    uint32_t *linked_list_node, DMA_TypeDef *dma)
+{
+	uint32_t next_desc = 1;
+	struct dma_block_config *block_config;
+	uint32_t registers_update = 0;
+	uint32_t addr_offset = 0;
+	uint32_t descriptor_index = 0;
+	uint32_t base_addr = 0;
+	uint32_t next_desc_addr = 0;
+	uint32_t channel;
+
+	block_config = config->head_block;
+	base_addr = (uint32_t)&linked_list_node[descriptor_index];
+
+	channel = dma_stm32_id_to_channel(id);
+	LL_DMA_SetLinkedListBaseAddr(dma, channel, base_addr);
+
+	for (int i = 0; i < config->block_count; i++) {
```

Comment:
Prefer use type `uint` since block_count is unsigned. Would prefer some compiler to complain on signed/unsigned comparison.